packages-content-model/roosterjs-content-model-api/lib/publicApi/utils/formatSegmentWithContentModel.ts

```diff
@@ -59,6 +59,15 @@ export function formatSegmentWithContentModel(
                 ? [[pendingFormat, null, null]]
                 : segmentAndParagraphs.map(item => [item[0].format, item[0], item[1]]);
 
+            //  If there is a pending format and the selection is collapsed, we need to apply the format to the selection marker either
+           if (pendingFormat && isCollapsedSelection) {
```

Comment:
Now I understand what you mean. I think a better fix should be when creating Content Model, directly set the format of SelectionMarker to be the pending one if it matches current pending format selection. Then all further cases where selection marker format is used will apply this result.

I can make a fix later. Or if you don't want to wait, please go ahead to make the change.

Reply:
I'm not in a hurry, since we cannot upgrade roosterjs for now. If you think this change makes sense, I can wait for your fix, and overrides the `formatSegmentWithContentModel` in our side as a workaround.